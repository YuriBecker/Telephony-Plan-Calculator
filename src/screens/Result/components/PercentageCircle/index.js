import React from 'react';
import { Text } from '@ui-kitten/components';
import ProgressCircle from 'react-native-progress-circle';
import PropTypes from 'prop-types';
import { ViewPropTypes, View } from 'react-native';
import colors from '../../../../styles/colors';

const index = ({ percentage, radius, style }) => {
  return (
    <View style={style}>
      <ProgressCircle
        percent={percentage}
        radius={radius}
        borderWidth={4}
        color={colors.primary}
        shadowColor={colors.grey}
        bgColor={colors.black[1]}
        style={style}
      >
        <Text
          style={{ fontSize: 20, color: colors.primary, fontWeight: 'bold' }}
        >
          {`${percentage}%`}
        </Text>
      </ProgressCircle>
    </View>
  );
};

export default index;

index.propTypes = {
  percentage: PropTypes.number.isRequired,
  radius: PropTypes.number,
  style: ViewPropTypes.style,
};

index.defaultProps = {
  radius: 50,
  style: {},
};
