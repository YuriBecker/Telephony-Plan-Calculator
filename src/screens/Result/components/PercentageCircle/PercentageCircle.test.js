import React from 'react';
import { shallow } from 'enzyme';

import PercentageCircle from './index';

describe('Testing PercentageCircle component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PercentageCircle percentage={37} />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
});
