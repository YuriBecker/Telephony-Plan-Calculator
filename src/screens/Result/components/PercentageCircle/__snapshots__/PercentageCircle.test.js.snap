// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PercentageCircle component should render correctly 1`] = `
<View
  style={Object {}}
>
  <PercentageCircle
    bgColor="#212121"
    borderWidth={4}
    color="#1db954"
    containerStyle={null}
    percent={37}
    radius={50}
    shadowColor="#b3b3b3"
    style={Object {}}
  >
    <TextComponent
      style={
        Object {
          "color": "#1db954",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      37%
    </TextComponent>
  </PercentageCircle>
</View>
`;
